passwords.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Passwords utilities.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const bcrypt = require('bcryptjs');
+
+module.exports = {
+  /**
+   * Hash a password asynchronously
+   * @param {String} password
+   * @return {Promise<String>} hashed password
+   */
+  hash: function(password) {
+    return bcrypt.hash(password, 10);
+  },
+
+  /**
+   * Hash a password synchronously
+   * @param {String} password
+   * @return {String} hashed password
+   */
+  hashSync: function(password) {
```

Comment:
^ hey do you mind adding a warning about how calling this will literally destroy the gateways ability to do things ? It's absolutely fine to do this when we boot up the db but otherwise it will put us in a bad spot until we have out of process adapters.